presto-hive/src/main/java/com/facebook/presto/hive/authentication/KerberosHadoopAuthentication.java

```diff
@@ -25,24 +27,22 @@
 public class KerberosHadoopAuthentication
         implements HadoopAuthentication
 {
-    static {
-        // In order to enable KERBEROS authentication method for HDFS
-        // UserGroupInformation.authenticationMethod static field must be set to KERBEROS
-        // It is further used in many places in DfsClient
-        Configuration configuration = new Configuration(false);
-        configuration.set("hadoop.security.authentication", "kerberos");
-        UserGroupInformation.setConfiguration(configuration);
-
-        // KerberosName#rules static field must be initialized
-        // It is used in KerberosName#getShortName which is used in User constructor invoked by UserGroupInformation#getUGIFromSubject
-        KerberosName.setRules("DEFAULT");
-    }
+    private static final Logger log = Logger.get(KerberosHadoopAuthentication.class);
 
     private final KerberosAuthentication kerberosAuthentication;
 
-    public KerberosHadoopAuthentication(KerberosAuthentication kerberosAuthentication)
+    public KerberosHadoopAuthentication(KerberosAuthentication kerberosAuthentication, HdfsConfiguration hdfsConfiguration)
     {
         this.kerberosAuthentication = requireNonNull(kerberosAuthentication, "kerberosAuthentication is null");
+        Configuration configuration = requireNonNull(hdfsConfiguration, "hdfsConfiguration is null").getConfiguration(null, null);
```

Comment:
https://github.com/prestodb/presto/pull/11077#discussion_r204056724

Reply:
my bad. Will fix it in the next update.